ln1_f4.py

```diff
@@ -1,115 +1,108 @@
-var_names_list = [cpu_total,cpu_user,cpu_sys,cpu_idle,mem_total,mem_used,mem_free,mem_cached,hdd_total,hdd_used,hdd_free]
-for x in var_names_list:
-    data_string.append(x)
+systime_customformat = datetime.datetime.now()
+systime_customformat = systime_customformat.strftime("%d,%m,%Y,%H,%M,%S")
 
-###############################################################################################################
-
-try:
-    database_file = open("database.txt", "r") #пробую відкрити файл БД
-    db_lines_count = int(database_file.readline())
+sysinfo_variables_names = [cpu_total, cpu_user, cpu_sys, cpu_idle, mem_total, mem_used, mem_free,
+                           mem_cached, hdd_total, hdd_used, hdd_free]
+sysinfo_database_keys = ["cpu_total", "cpu_user", "cpu_sys", "cpu_idle", "mem_total", "mem_used", "mem_free",
+                         "mem_cached", "hdd_total", "hdd_used", "hdd_free"]
 
-except IOError:
-    database_file = open("database.txt", "w") #якщо помилка то створюю чистий
-    database_file.write("0"+"\n")
-    db_lines_count = 0
-
-finally:
-    database_file.close() #закриваю файл
+###############################################################################################################
 
-database_file = open("database.txt", "a") #відкриваю файл та добавляю рядок з данними в БД
-database_file.write(str(data_string) + "\n")
-database_file.close()
-db_lines_count += 1
-database_file = open("database.txt", "r+") #перезаписую лічильник кількості рядків в БД
-database_file.seek(0,0)
-database_file.write(str(db_lines_count) + "\n")
-database_file.close()
+if os.path.isfile("database2") == False:
+    with open("database2", "w") as database_file:
+        sysinfo_database = {}
+        database_file.write(json.dumps(sysinfo_database))
+
+with open("database2", "r") as database_file:
+    data = database_file.read().strip()
+sysinfo_database = json.loads(data)
+sysdb_totalrecords = len(sysinfo_database) + 1
+sysinfo_database[str(sysdb_totalrecords)] = {
+    "record_time": systime_customformat,  # saving time but not using at the moment (plan in future)
+    "cpu_total": cpu_total,
+    "cpu_user": cpu_user,
+    "cpu_sys": cpu_sys,
+    "cpu_idle": cpu_idle,
+    "mem_total": mem_total,
+    "mem_used": mem_used,
+    "mem_free": mem_free,
+    "mem_cached": mem_cached,
+    "hdd_total": hdd_total,
+    "hdd_used": hdd_used,
+    "hdd_free": hdd_free
+}
+with open("database2", "w") as database_file:
+    database_file.write(json.dumps(sysinfo_database))
 
 ###############################################################################################################
 
-if db_lines_count >= 12: #перевіряю чи записана достатня кількість рядків, якщо так то починаю обробку
-    database_file = open("database.txt", "r")
-    dump = database_file.readline() #"пуста" процедура зчитування щоб пропустити рядок з лічильником
+if sysdb_totalrecords >= 4:  # check if batabase has enough records
```

Comment:
4 in not ment to limit writing into database, it's to check if database has at least 4 record befor program starts to averge, send emails and etc. As for separate averaging function i'll redo this part